File: backend/app/schemas/consultation_image.py
"""
Consultation Image Schemas - Pydantic models for image API
"""

from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
from typing import Optional


class ConsultationImageBase(BaseModel):
    """Base schema for consultation image"""
    image_type: Optional[str] = Field(None, max_length=100, description="Type d'image (lésion primaire, macro, dermatoscope, etc.)")
    body_location: Optional[str] = Field(None, max_length=200, description="Localisation anatomique de la lésion")
    description: Optional[str] = Field(None, description="Description médicale de l'image")
    is_primary: bool = Field(False, description="Image principale de la consultation")


class ConsultationImageCreate(ConsultationImageBase):
    """Schema for creating a new consultation image (without file upload)"""
    consultation_id: int = Field(..., gt=0, description="ID de la consultation")


class ConsultationImageUpdate(BaseModel):
    """Schema for updating image metadata"""
    image_type: Optional[str] = Field(None, max_length=100)
    body_location: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    is_primary: Optional[bool] = None


class ConsultationImageMetadata(BaseModel):
    """Schema for EXIF metadata"""
    captured_at: Optional[datetime] = None
    camera_model: Optional[str] = None
    width: Optional[int] = None
    height: Optional[int] = None


class ConsultationImageResponse(ConsultationImageBase):
    """Schema for consultation image response"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    consultation_id: int
    image_url: str
    thumbnail_url: Optional[str] = None
    original_filename: str
    file_size: int
    mime_type: str
    width: Optional[int] = None
    height: Optional[int] = None
    captured_at: Optional[datetime] = None
    camera_model: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class ConsultationImageListResponse(BaseModel):
    """Schema for paginated list of consultation images"""
    images: list[ConsultationImageResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class ImageUploadResponse(BaseModel):
    """Schema for successful image upload response"""
    id: int
    consultation_id: int
    image_url: str
    thumbnail_url: Optional[str] = None
    original_filename: str
    file_size: int
    mime_type: str
    message: str = "Image uploaded successfully"

File: frontend/src/types/consultation-image.ts
/**
 * Consultation Image Types
 * Types for medical image management in consultations
 */

export interface ConsultationImage {
  id: number
  consultation_id: number
  image_url: string
  thumbnail_url: string | null
  original_filename: string
  file_size: number
  mime_type: string
  width: number | null
  height: number | null

  // Medical metadata
  image_type: string | null
  body_location: string | null
  description: string | null
  is_primary: boolean

  // EXIF data
  captured_at: string | null
  camera_model: string | null

  // Timestamps
  created_at: string
  updated_at: string
}

export interface ConsultationImageListResponse {
  images: ConsultationImage[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface ImageUploadResponse {
  id: number
  consultation_id: number
  image_url: string
  thumbnail_url: string | null
  original_filename: string
  file_size: number
  mime_type: string
  message: string
}

export interface ConsultationImageUpdate {
  image_type?: string
  body_location?: string
  description?: string
  is_primary?: boolean
}

export interface ImageUploadFormData {
  file: File
  image_type?: string
  body_location?: string
  description?: string
  is_primary?: boolean
}

// Helper types for UI
export interface ImagePreview {
  file: File
  preview: string
  metadata: {
    size: number
    type: string
    name: string
  }
}
